src/frontend/komponenter/Oppgaver/OppgaveList.tsx

```diff
@@ -28,223 +29,247 @@ interface IOppgaveListProps {
     innloggetSaksbehandler?: ISaksbehandler;
 }
 
+enum ListFelt {
+    OPPRETTET_TIDSPUNKT = 'OPPRETTET_TIDSPUNKT',
+    OPPGAVETYPE = 'OPPGAVETYPE',
+    BEHANDLINGSTEMA = 'BEHANDLINGSTEMA',
+    FRIST_FERDIGSTILLELSE = 'FRIST_FERDIGSTILLELSE',
+    PRIORITET = 'PRIORITET',
+    AKTOER_ID = 'AKTOER_ID',
+    TILDELT_ENHETSNR = 'TILDELT_ENHETSNR',
+    TILORDNET_RESSURS = 'TILORDNET_RESSURS',
+    HANDLINGER = 'HANDLINGER',
+    BESKRIVELSE = 'BESKRIVELSE',
+}
+
+enum FeltSortOrder {
+    NONE = 'none',
+    ASCENDANT = 'ascendant',
+    DESCENDANT = 'descendant',
+}
+
+const feltLabelMap = new Map<ListFelt, string>([
+    [ListFelt.OPPRETTET_TIDSPUNKT, 'Reg. dato'],
+    [ListFelt.OPPGAVETYPE, 'Oppgavetype'],
+    [ListFelt.BEHANDLINGSTEMA, 'Gjelder'],
+    [ListFelt.FRIST_FERDIGSTILLELSE, 'Frist'],
+    [ListFelt.PRIORITET, 'Prioritet'],
+    [ListFelt.AKTOER_ID, 'Bruker'],
+    [ListFelt.TILDELT_ENHETSNR, 'Enhet'],
+    [ListFelt.TILORDNET_RESSURS, 'Saksbehandler'],
+    [ListFelt.HANDLINGER, 'Handlinger'],
+    [ListFelt.BESKRIVELSE, 'Beskrivelse'],
+]);
+
+const oppgaveFeltMap = new Map<ListFelt, string>([
+    [ListFelt.OPPRETTET_TIDSPUNKT, 'opprettetTidspunkt'],
+    [ListFelt.OPPGAVETYPE, 'oppgavetype'],
+    [ListFelt.BEHANDLINGSTEMA, 'behandlingstema'],
+    [ListFelt.FRIST_FERDIGSTILLELSE, 'fristFerdigstillelse'],
+    [ListFelt.PRIORITET, 'prioritet'],
+    [ListFelt.AKTOER_ID, 'aktoerId'],
+    [ListFelt.TILDELT_ENHETSNR, 'tildeltEnhetsnr'],
+    [ListFelt.TILORDNET_RESSURS, 'tilordnetRessurs'],
+    [ListFelt.HANDLINGER, 'handlinger'],
+    [ListFelt.BESKRIVELSE, 'beskrivelse'],
+]);
+
+const ariaSortMap = new Map<FeltSortOrder, 'none' | 'descending' | 'ascending'>([
+    [FeltSortOrder.NONE, 'none'],
+    [FeltSortOrder.DESCENDANT, 'descending'],
+    [FeltSortOrder.ASCENDANT, 'ascending'],
+]);
+
+const sortLenkClassNameMap = new Map<
+    FeltSortOrder,
+    '' | 'tabell__th--sortert-desc' | 'tabell__th--sortert-asc'
+>([
+    [FeltSortOrder.NONE, ''],
+    [FeltSortOrder.DESCENDANT, 'tabell__th--sortert-desc'],
+    [FeltSortOrder.ASCENDANT, 'tabell__th--sortert-asc'],
+]);
+
+const initFeltOrder = new Map<ListFelt, FeltSortOrder>([
+    [ListFelt.OPPRETTET_TIDSPUNKT, FeltSortOrder.NONE],
+    [ListFelt.OPPGAVETYPE, FeltSortOrder.NONE],
+    [ListFelt.BEHANDLINGSTEMA, FeltSortOrder.NONE],
+    [ListFelt.FRIST_FERDIGSTILLELSE, FeltSortOrder.NONE],
+    [ListFelt.PRIORITET, FeltSortOrder.NONE],
+    [ListFelt.TILDELT_ENHETSNR, FeltSortOrder.NONE],
+    [ListFelt.TILORDNET_RESSURS, FeltSortOrder.NONE],
+]);
```

Comment:
Nice, but move this to a types.ts file, perhaps oppgaveFilter.ts?

Reply:
Is it ok to call the file OppgaveListFelt.ts?

Reply:
Sure 👍